xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/email/live/LiveNotificationEmailListener.java

```diff
@@ -0,0 +1,167 @@
+     */
+    public static final String NAME = "Live Notification Email Listener";
+
+    @Inject
+    private RecordableEventDescriptorManager recordableEventDescriptorManager;
+
+    @Inject
+    private LiveNotificationEmailManager liveNotificationEmailManager;
+
+    @Inject
+    private NotificationConfiguration notificationConfiguration;
+
+    @Inject
+    @Named("context")
+    private ComponentManager componentManager;
+
+    @Inject
+    private Logger logger;
+
+    private Thread notificationGraceTimeThread;
+
+    /**
+     * Thread used for triggering {@link LiveNotificationEmailManager#run()} when needed (ie : when an event
+     * grace time has ended).
+     */
+    private class NotificationGraceTimeRunnable implements Runnable
+    {
+        @Override
+        public void run()
+        {
+            DateTime nextWakeUpTime = DateTime.now();
+
+            // If the nextWakeUpTime is null, the thead will die, but will be restarted by
+            // LiveNotificationEmailListener#onEvent(Event, Object, Object) as soon as a new event is caught.
+            while (nextWakeUpTime != null) {
+                if (nextWakeUpTime.isAfterNow()) {
+                    try {
+                        Thread.sleep(nextWakeUpTime.getMillis() - DateTime.now().getMillis());
+                    } catch (IllegalArgumentException | InterruptedException e) {
+                        if (e instanceof InterruptedException) {
+                            return;
+                        }
+                    }
+                }
+
+                liveNotificationEmailManager.run();
+
+                nextWakeUpTime = liveNotificationEmailManager.getNextExecutionDate();
+            }
+        }
+    }
+
+    /**
+     * Constructs a new {@link LiveNotificationEmailListener}.
+     */
+    public LiveNotificationEmailListener()
+    {
+        super(NAME, new EventStreamAddedEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object o, Object o1)
+    {
+        try {
+
+            org.xwiki.eventstream.Event eventStreamEvent = (org.xwiki.eventstream.Event) o;
+
+            // We can’t directly store a list of RecordableEventDescriptors as some of them can be
+            // dynamically defined at runtime.
+            List<RecordableEventDescriptor> descriptorList =
+                    this.recordableEventDescriptorManager.getRecordableEventDescriptors(true);
+
+            // Try to match one of the given descriptors with the current event.
+            for (RecordableEventDescriptor descriptor : descriptorList) {
+                // Find a descriptor that corresponds to the given event
+                // We also check if the notifications are enabled in the wiki and if the mail option for the
+                // notifications is enabled.
+                if (descriptor.getEventType().equals(eventStreamEvent.getType())
+                        && this.notificationConfiguration.isEnabled()
```

Comment:
Do this check before the for(), because if `this.notificationConfiguration.isEnabled()` or `this.notificationConfiguration.areEmailsEnabled()` are false, you don't event need to load the descriptors and so on...